### DISCUSSION TITLE
contract: update applyToRound signature

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#171] [Q&A] [answered]
Currently the signature is 
`applyToRound(address project,  MetaPtr metaPtr);`

This can be found here: https://github.com/gitcoinco/grants-round/blob/main/packages/contracts/contracts/round/RoundImplementation.sol#L251

The grants hub team mentioned that every project does not have it's own contract aka no unique identifier.
It still makes sense for us to have a unique id when the project applies -> to make it easier to index via subgraph 

We have two means of going about this which would require making contract changes.

**Option 1:**
 change address to bytes32. The grant hub can decide how to generate the unique hash of the project. The goal is to make sure this is unique across registries across all chains,
 
 `applyToRound(bytes32 projectID,  MetaPtr metaPtr);`

**Option 2:**
 change address to string. The grant hub can decide how to generate the unique hash of the project but this gives us the option to add a prefix / suffix if needed
 
 `applyToRound(string projectID,  MetaPtr metaPtr);`

**Comment by QEDK** [comment]
I would argue for `Option 1`, it ties in well with randomly created `keccak256` hashes on-chain, like `keccak256(abi.encode(<user input string>, <serial id>, block.number))` Not to mention gas-cost savings!

**Reply by thelostone-mc** [reply]
+1 on this .

**Comment by thelostone-mc** [comment]
@nfrgosselin we should have this added to the backlog as this is contract change which break things in a few places on the hub side -> the longer we delay it

**Comment by gravityblast** [comment, marked as answer]
should the Round contract be aware of the `ProjectRegistry` contract?
If it knows about it, and a project owner applies with `project 1`, the Round contract can check if the `msg.sender` is actually a project owner. 
Otherwise anyone can apply with any project even if they don't own it, and set any recipient address in the application metadata.

**Comment by vacekj** [comment]
Closing as stale.
